### PR TITLE
[bshoshany-thread-pool] Updated to v3.4.0

### PR DESCRIPTION
**Pull request by bshoshany**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Thanks!

### CODE REVIEW

**Review by Adela0814 — DISMISSED** [review]

File: ports/bshoshany-thread-pool/portfile.cmake [hunk 1]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO bshoshany/thread-pool
-    REF v3.3.0
-    SHA512 980c8dfe90e04cbc622ee4ae2ce67444069311619f0dc5d7ac3b3a3ea59ead42d4c867e669e8ce9c71cdc2a4bae431402a8f5032ab29fdfc1ca507b0e7102d8f
+    REF v3.4.0
```
Comment (Adela0814) [inline ports/bshoshany-thread-pool/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"

```

**Comment by Adela0814** [comment]
@bshoshany Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Comment by bshoshany** [comment]
Thanks! I accepted your changes, but now it's failing a check :(

File: ports/bshoshany-thread-pool/portfile.cmake [hunk 2]
```diff
@@ -1,8 +1,9 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO bshoshany/thread-pool
-    REF v3.3.0
-    SHA512 980c8dfe90e04cbc622ee4ae2ce67444069311619f0dc5d7ac3b3a3ea59ead42d4c867e669e8ce9c71cdc2a4bae431402a8f5032ab29fdfc1ca507b0e7102d8f
+    REF "v${VERSION}"
+
```
Comment (Adela0814) [inline ports/bshoshany-thread-pool/portfile.cmake:5]: Please remove this line.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by bshoshany** [comment]
I removed the line you indicated, but it's still failing a check - although maybe I didn't understand what you wanted me to change. Perhaps we should just revert to the original version? Thanks.

**Comment by Adela0814** [comment]
@bshoshany  The CI error was reported because the SHA value is incorrect. After making the changes, you should run the command again: `\vcpkg.exe x-add-version bshoshany-thread-pool --overwrite-version`.

**Comment by bshoshany** [comment]
In the end I decided to just revert to my original version, since unfortunately I don't have more time to devote to this PR. I will incorporate the `REF "v${VERSION}"` trick next time I update the package. Thanks again :)

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**